vulture_os/authentication/openid/views.py

```diff
@@ -85,13 +88,28 @@ def edit(request, object_id=None, api=False):
         form = OpenIDRepositoryForm(request.POST or None, instance=repo, error_class=DivErrorList)
 
     if request.method in ("POST", "PUT") and form.is_valid():
+        old_jwt_key_filename = repo.get_jwt_key_filename()
         # Save the form to get an id if there is not already one
         repo = form.save(commit=False)
         # If provider_url changed, force reload of configuration
         if "provider_url" in form.changed_data:
             repo.last_config_time = None
+        if "jwt_key" in form.changed_data and repo.jwt_signature_type not in ("HS256", "HS384", "HS512"):
+            Cluster.api_request("system.config.models.delete_conf", old_jwt_key_filename)
+            repo.save_conf()
         repo.save()
 
+        portal = UserAuthentication.objects.get(name=repo.name.replace("Connector_", ""))
+        portal.save_conf()
```

Comment:
Careful! an OpenID Connector is not always linked to a Vulture IDP!

Reply:
You're right ! I'll add an if exists statement

Reply:
Conf files refreshing of the related IDP is no longer needed